servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -144,6 +144,7 @@
     protected int domainNameMaxLen;
     protected AuthorizedServices serverAuthorizedServices = null;
     protected SolutionTemplates serverSolutionTemplates = null;
+    protected Map<String, Integer> eligibleTemplatesForAutoUpdate = null;
```

Comment:
this is needed because we are populating this map via load solution templates during early stage of app startup and using it in autoApplyTemplate() in the later stages. also we need to load object store in between which is required to called applytemplate function from dbservice. so there is no other way to pass this data down to autoApplyTemplate() function.

Reply:
will move the code block and remove the field reference from class.

Reply:
> Please also make sure to run openclover at the end and make sure all new lines of code are covered with test cases:
> 
> mvn -P coverage clean org.openclover:clover-maven-plugin:instrument org.openclover:clover-maven-plugin:clover

yes did this and added more tests to cover few missing lines.